packages/react-server-dom-vite/src/shared/ReactFlightImportMetadata.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+export type ImportManifestEntry = {
+  id: string,
+  // chunks is an array of filenames
+  chunks: Array<string>,
+  name: string,
+};
+
+// This is the parsed shape of the wire format which is why it is
+// condensed to only the essentialy information
+export type ImportMetadata =
+  | [
+      /* id */ string,
+      /* chunk filenames */ Array<string>,
+      /* name */ string,
+      /* async */ 1,
+    ]
+  | [/* id */ string, /* chunk filenames */ Array<string>, /* name */ string];
```

Comment:
I don't know a whole lot about the dev/prod differences for vite but I thought dev was ESM and prod was chunks via esbuild or something. I would expect the wire format for RSC to be different depending on whether it was dev or prod